### ISSUE TITLE
I think  it is a great work !Can you share your train code and dataset?

### ISSUE THREAD

**Issue by c1a1o1** [minyoungg/selfconsistency#2]
I think  it is a great work !Can you share your train code and dataset?
Thank you very much!

**Comment by AndrewHLiu** [comment]
Talking with others to figure out the best way to release this section. Thanks for asking!

**Comment by minyoungg** [comment]
We sort our data into a dictionary .. where each key is an EXIF tag that is mapped to N lists (N is the number of values for the given EXIF tag). Each list consists of all the images that has this value. Hence, we can easily sample a batch from these lists. We assign a null label (e.g -1) to the images that either does not have a tag or is under the pruning threshold. When you train your model you can either mask the loss or always assign a target of 0 when one of the label is a null value.

**Comment by ChrisLab91** [comment]
Thanks so much for your answer!

I think there is one more thing that is unclear to me:

Did you sample each picture pair for one (random?) "tag" and selected a value for that tag uniformly? Or did you rather select the pictures such that you also gurantee 50% of each batch shares the value for EVERY (used) tag. It would be not trivial to me how to achieve such an selection.

**Comment by minyoungg** [comment]
For each training batch we choose 1 exif tag and 1 value within that tag. These are chosen uniformly at random. Given this tag and value we make sure half of the batch will be consistent (both images have the same value) and half of the batch will be inconsistent (1 image will have the value and the other will not). The images are chosen at random from the list explained above. We do this because it is computationally expensive to find an unbiased set of images. Of course this introduces biases on other exif tags if we compute loss on all the tags and backprop on it. Although we did not do this in our original implementation, one could just backprop on that specific tag to reduce bias ... which I would assume prolongs the training time. Hope this helps.

**Comment by ChrisLab91** [comment]
Thanks!

**Comment by yelusaleng** [comment]
@c1a1o1 @ChrisLab91 
hey guys,  have you met the `Intel MKL ERROR: Parameter 6 was incorrect on entry to DLASWP` when you run ncuts_demo.py ?  could you give me some solving suggestions?

**Comment by shgidi** [comment]
Hi, I would also find it really helpful if you could add a training example.
Many thanks!

**Comment by imjiangjun** [comment]
A great work! And I'm really looking forward to reproducing the results. 
I'm very appreciate it if you could provide the dataset and trainning examples.

**Comment by shauryagoel0** [comment]
Hi, I am still not clear about some things-
1) after choosing the tag and the value uniformly for a batch, while constructing the inconsistent half batch, for each pair of images, one image has the same value as the value for the consistent images while the other image has a different value. Do they have the same tag but a different value or it can have different tags? 
2) we are computing loss for whether the chosen tag was same or not. Do we check whether a consistent pair of images have some other tag which has same value besides the chosen tag (like in fig.3 of the paper)? If so, which loss function is used?
3) how to compute the loss of inconsistent image pairs?

**Comment by kuznetsoffandrey** [comment]
Can you please share the trained weights zip? It is unavailable on berkley website

**Comment by vincenzocolella** [comment]
Hi ! It would be amazing if you could share the training implementation on github. Thanks a lot !

**Comment by onaidra** [comment]
Nice job! is it possible to have a main in order to train on another dataset? 

thanks in advance!

**Comment by onaidra** [comment]
i'm having problems in exif_solver function setup_data because i can't understand nextbatch at which data type is referred to.
Another problem that i'm struggling to understand is the functions' order used to train the dataset.

Any suggestion?

**Comment by onaidra** [comment]
@minyoungg @AndrewHLiu  I am not understanding what setup_data(self, data, data_fn=None) is expecting as input, because data_fn in get_data is used as a function "self.data_fn(batch_size,split=split)" but there is no function with input split

**Comment by onaidra** [comment]
@ChrisLab91 have you been able to train on another dataset?

**Comment by christian-holland** [comment]
> @ChrisLab91 have you been able to train on another dataset?

Yes, I downloaded it from flickr myself they have / had relatively mild regulations on downloads if you have the bandwidth. I don't have the dataset anymore though.

**Comment by onaidra** [comment]
@christian-holland thanks for your reply, can you please contact me privately on my email address? 
I just wanted some advice in how to create a "main" in order to train the dataset.

Thanks in advance
